src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28928,6 +28928,159 @@ public void Init(T t)
                 Diagnostic(ErrorCode.WRN_MemberNotNull, "}").WithArguments("field").WithLocation(18, 5));
         }
 
+        [Fact, WorkItem(48996, "https://github.com/dotnet/roslyn/issues/48996")]
+        public void DisallowNullProperty_Constructor()
+        {
+            var source = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+public class C1
+{
+    [DisallowNull, NotNull] public string? F { get; set; }
+
+    public C1() { } // 1
+    public C1(string? F) // 2
+    {
+        this.F = F; // 3
+    }
+
+    public C1(int i)
+    {
+        this.F = i.ToString();
+    }
+}
+
+public class C2
+{
+    [NotNull] public string? F { get; set; }
+
+    public C2() { } // 1
+    public C2(string? F)
+    {
+        this.F = F;
+    }
+
+    public C2(int i)
+    {
+        this.F = i.ToString();
+    }
+}
+
+public class C3
+{
+    [DisallowNull] public string? F { get; set; }
+
+    public C3() { }
+    public C3(string? F)
+    {
+        this.F = F; // 1
+    }
+
+    public C3(int i)
+    {
+        this.F = i.ToString();
+    }
+}
+";
+            // Note that while lack of warnings on `C2(string?)` seems strange, it is expected
+            // 'C2.F' begins in a maybe-null state, but assigning maybe-null to it causes it to become not-null
```

Comment:
Please elaborate this sentence a bit. After I thought about it for a couple minutes I figured out what you meant, but the fact that it becomes non-null because it's a property and the getter and setter are not linked is not obvious.

Reply:
Have updated the comment, please let me know if you think it's sufficiently clear.

Reply:
I actually think it's less clear now, sorry. It's simply that the getter and setter are not linked: regardless of what we assign in the setter, the getter will return `NotNull` because that's how the property is annotated.

Reply:
> It's simply that the getter and setter are not linked

I'm not sure I understand what linked means here. Are a property's accessors "linked" when the flow state of the property after assignment is the same as the flow state of the 'value' argument to the property setter?

I guess when I hear "not linked" I imagine a property setter which never changes the flow state from the getter. However, that is not the case here.

The language I used in this comment is how we summarized this behavior when it was designed. However, I can see that not all the context is given for it here, so I'm open to revising it again.

Reply:
It's not linked in that the type of the thing put into the setter is not reflected in (ie, not linked to) the flow state of the getter.